components/types/fulfillment.ts

```diff
@@ -1,8 +1,15 @@
+import type { CartItem } from "@/server/types/GraphQL"
+
 // fulFillment
 export type Fulfillment = {
-  name: string
-  value: string
-  label: string
-  details: string
-  required: boolean
+  name?: string
+  value?: string
+  label?: string
+  details?: string
+  required?: boolean
+  code?: string
+  shortName?: string
+  disabled?: boolean
+  cartItem?: CartItem
```

Comment:
why is cart item now a property of this?

Reply:
Changed to pass specific properties that are needed in the components.